jts-test-runner/src/lib.rs

```diff
@@ -17,24 +21,78 @@ pub fn assert_jts_tests_succeed(pattern: &str) {
 
     // sanity check that *something* was run
     assert!(
-        runner.failures().len() + runner.successes().len() > 0,
+        runner.unexpected_failures().len() + runner.successes().len() > 0,
         "No tests were run."
     );
 
-    if !runner.failures().is_empty() {
+    if !runner.unexpected_failures().is_empty() {
         let failure_text = runner
-            .failures()
+            .unexpected_failures()
             .iter()
             .map(|failure| format!("{}", failure))
             .collect::<Vec<String>>()
             .join("\n");
+
         panic!(
-            "{} failures / {} successes in JTS test suite:\n{}",
-            runner.failures().len(),
+            "{} unexpected failures / {} successes in JTS test suite:\n{}",
+            runner.unexpected_failures().len(),
             runner.successes().len(),
             failure_text
         );
     }
+
+    for success in runner.successes() {
+        if runner.expected_failures().contains(&success.test_id) {
+            panic!("Test {:?} was expected to fail, but it passed. Did you fix something? Good job! Update `expected_failures`.", success.test_id);
+        }
+    }
+}
+
+pub fn check_buffer_test_case(
+    actual: &Geometry,
+    expected: &Geometry,
+) -> std::result::Result<(), String> {
+    // This error threshold is arbitrary. I ratcheted it down until tests started failing.
+    // Manually inspecting the output of the borderline cases appeared subjectively "reasonable".
+    //
+    // In particular, we seem to diverge the most when doing large subtractive (negative buffers)
```

Comment:
Is there a particular test case that showed this difference? Could be useful to preserve somewhere

Reply:
You can ratchet this tolerance down and see which tests start to fail.

Here's an example just below the current tolerance:

<img width="452" alt="Screenshot 2025-06-17 at 12 13 51" src="https://github.com/user-attachments/assets/59e07c34-b76e-4e53-a9e1-6f891afcd0cb" />

The blue is the input, and the yellow is the output of negative buffering. We get a similar output from our implementation, but there is a bit of divergence (just under the tolerance).

Reply:
Seems very very negligible